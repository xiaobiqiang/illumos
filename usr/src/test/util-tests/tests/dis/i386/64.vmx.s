/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test 64-bit specific VMX related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vmclear		(%rax)
	vmptrld		(%rax)
	vmptrst		(%rax)
	vmread		%rbx, %rax
	vmread		%rbx, (%rax)
	vmreadq		%rbx, %rax
	vmreadq		%rbx, (%rax)
	vmwrite		%rax, %rbx
	vmwrite		(%rax), %rbx
	vmwriteq	%rax, %rbx
	vmwriteq	(%rax), %rbx
.size libdis_test, [.-libdis_test]
