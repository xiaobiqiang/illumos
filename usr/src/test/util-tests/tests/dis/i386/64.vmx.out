    libdis_test:      66 0f c7 30        vmclear (%rax)
    libdis_test+0x4:  0f c7 30           vmptrld (%rax)
    libdis_test+0x7:  0f c7 38           vmptrst (%rax)
    libdis_test+0xa:  0f 78 d8           vmread %rbx,%rax
    libdis_test+0xd:  0f 78 18           vmread %rbx,(%rax)
    libdis_test+0x10: 0f 78 d8           vmread %rbx,%rax
    libdis_test+0x13: 0f 78 18           vmread %rbx,(%rax)
    libdis_test+0x16: 0f 79 d8           vmwrite %rax,%rbx
    libdis_test+0x19: 0f 79 18           vmwrite (%rax),%rbx
    libdis_test+0x1c: 0f 79 d8           vmwrite %rax,%rbx
    libdis_test+0x1f: 0f 79 18           vmwrite (%rax),%rbx
